pkg/gardenlet/controller/seed/seed.go

```diff
@@ -208,6 +220,61 @@ func (c *Controller) Run(ctx context.Context, workers int) {
 	waitGroup.Wait()
 }
 
+func (c *Controller) startHealthManagement(ctx context.Context) {
+	var (
+		seedName              = confighelper.SeedNameFromSeedConfig(c.config.SeedConfig)
+		seedLabelSelector     labels.Selector
+		expectedHealthReports int
+		err                   error
+	)
+
+	if seedName != "" {
+		expectedHealthReports = 1
+	} else {
+		seedLabelSelector, err = metav1.LabelSelectorAsSelector(c.config.SeedSelector)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	for {
```

Comment:
Now with this for loop, do you think it is still relevant to use a `periodicHealthz`? Seems like we already have a reset mechanism with `c.leaseMap = make(map[string]bool)` here.

The reason I ask is because of the time intervals in different go routines we're about to introduce: `periodicHealthz` (20 sec.) - this loop (10 sec.) - `seed_lease_control` (2 sec.). Maybe we cloud simplify this handling if we went w/o `periodicHealthz`.

Reply:
I think it's still relevant, yes. What if this controller breaks/blocks for some reason? We still want the `/healthz` to be set to `false` in this case.

Reply:
Okay, I'm not quite sure how this can cause an issue since we are doing it in a Go routine but also no objections against introducing it for safety reason 👍 Thanks for letting know.

Reply:
Yeah, I agree, it's not a very good example, but I think it's still safer to let it automatically reset by an independent mechanism.